docs/configuration.md

```diff
@@ -156,6 +156,13 @@ of the most common options to set are:
   </td>
 </tr>
 <tr>
+ <td><code>spark.executor.instances</code></td>
```

Comment:
Thanks for reviewing this PR, @srowen 

Actually I am confused by this. From the information below, it should not be used for yarn only:

- https://github.com/apache/spark/blob/master/core/src/test/scala/org/apache/spark/SparkContextSuite.scala#L403
- In `configuration.md`:
```
<tr>
  <td><code>spark.dynamicAllocation.initialExecutors</code></td>
  <td><code>spark.dynamicAllocation.minExecutors</code></td>
  <td>
    Initial number of executors to run if dynamic allocation is enabled.
    <br /><br />
    If `--num-executors` (or `spark.executor.instances`) is set and larger than this value, it will
    be used as the initial number of executors.
  </td>
</tr>
```   

- Currently`spark.executor.instances` is defined in [config/package.scala](https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/internal/config/package.scala#L86). If it is for YARN-specific, it should be defined in `yarn/config.scala` instead of `config/package.scala`.